workspaces/ui-v2/src/spectacle-implementations/cloud-viewer.tsx

```diff
@@ -145,14 +139,78 @@ class CloudInMemorySpectacle
   }
 }
 
-export interface InMemoryBaseSpectacle extends IBaseSpectacle {
+export interface CloudInMemoryBaseSpectacle extends IBaseSpectacle {
   samples: any[];
   opticContext: IOpticContext;
 }
 
+export interface CloudInMemorySpectacleDependencies {
+  events: any[];
+  samples: any[];
+}
+
+export type CloudInMemorySpectacleDependenciesLoader = () => Promise<CloudInMemorySpectacleDependencies>;
+
+//@SYNC: useInMemorySpectacle
 export function useCloudInMemorySpectacle(
   loadDependencies: CloudInMemorySpectacleDependenciesLoader
-): AsyncStatus<InMemoryBaseSpectacle> {
-  //@dev fill this in
-  throw new Error('copy me from public-examples.tsx when ready');
+): AsyncStatus<CloudInMemoryBaseSpectacle> {
+  const opticEngine = useOpticEngine();
+  const [spectacle, setSpectacle] = useState<CloudInMemoryBaseSpectacle>();
+  const [inputs, setInputs] = useState<{
+    events: any[];
+    samples: any[];
+  } | null>(null);
+
+  useEffect(() => {
```

Comment:
Does opticengine ever change? If not, it might be simpler to consolidate these useEffects (such that await loadDependencies -> opticContext load from that)? I see this being useful only if opticEngine changes (which it doesn't look like the current implementation does)

Reply:
The goal is to have the initial state loaded async, and from then on, update so there's a new spectacle instance every time the events change. I imagine we'll revisit this as part of our ui state management cleanup